libraries/AP_Scripting/AP_Scripting.cpp

```diff
@@ -97,12 +97,11 @@ void AP_Scripting::init(void) {
     }
 
     const char *dir_name = SCRIPTING_DIRECTORY;
-
-    if (!AP::FS().mkdir(dir_name)) {
-        gcs().send_text(MAV_SEVERITY_INFO, "Lua: created new directory (%s)", dir_name);
-    } else if (errno != EEXIST) {
-        gcs().send_text(MAV_SEVERITY_INFO, "Lua: failed to create (%s)", dir_name);
-        return;
+    if (AP::FS().mkdir(dir_name)) {
```

Comment:
Did you lose the `!` here?

Reply:
@peterbarker, I think it's correct as it is... either that or it was always printing it backwards.

Reply:
Yep.  Thought mkdir may have gone bool as part of the abstraction.

Reply:
... it should probably say `!=0`, it occurs to me, to make things clearer.

Reply:
@peterbarker, it's super weird that mkdir returns 0 if it works.  I guess that's a linux-y thing..